app/src/main/java/com/example/bigowlapp/activity/ViewAnotherUserActivity.java

```diff
@@ -1,112 +1,133 @@
+    String supBtnAlready = "You are supervising this user";
+    String supBtnCancel = "You have sent a request already. Cancel request";
+    String canNotSend = "Can not send ";
     private Button supRequestBtn;
-    private final FirebaseFirestore db = FirebaseFirestore.getInstance();
-    String otherUserNumber = getIntent().getStringExtra("smsNumber");
-    /*String contactNumber = getIntent().getStringExtra("contactNumber");*/
+    private AuthRepository authRepository = new AuthRepository();
+    String requestUID;
+    private TextView noteTv;
+    private TextView noteTv2;
+    String canCancel = "You currently have a pending request to supervise this user";
+    String noRequest = "You presently have NO request to supervise this user ";
+    String noSelfRequest = "This contact matches your contact. You can't send a request to yourself";
+    String superviseAlready = "You already have an accepted request to supervise this user";
+
+    NotificationRepository notificationRepository = new NotificationRepository();
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_view_another_user);
-
+        otherUser = getIntent().getParcelableExtra("user");
+        otherUserID = otherUser.getUId();
         supRequestBtn = findViewById(R.id.SupRequest);
-        supRequestRef = FirebaseDatabase.getInstance().getReference().child("SupRequests");
+        currentUserID = authRepository.getCurrentUser().getUid();
+        noteTv = findViewById(R.id.note);
+        noteTv = findViewById(R.id.note);
+        noteTv2 = findViewById(R.id.note2);
+        String contactDetails = getIntent().getStringExtra("contactDetails");
+        noteText = "Contact: " + contactDetails + " is already registered to the application.";
+        noteTv.setText(noteText);
 
-        db.collection("users")
-                .whereEqualTo("phoneNumber", otherUserNumber)
-                .get()
-                .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+        if (currentUserID.equals(otherUserID)) {
+            noteTv2.setText(noSelfRequest);
+            supRequestBtn.setClickable(false);
+            supRequestBtn.setText(canNotSend);
+        } else {
+            supRequestBtn.setClickable(true);
+            try {
+                observeRequests();
+                supRequestBtn.setOnClickListener(new View.OnClickListener() {
                     @Override
-                    public void onComplete(@NonNull Task<QuerySnapshot> task) {
-                        if (task.isSuccessful()) {
-                            task.getResult();
-                            for (QueryDocumentSnapshot qds : task.getResult()) {
-                                otherUser = qds.toObject(User.class);
-                            }
-
-                        }
+                    public void onClick(View v) {
```

Comment:
Please make this a lambda function as noted by SonarCloud

Reply:
It has been changed